src/implementation/algorithm.rs

```diff
@@ -244,13 +245,26 @@ macro_rules! algorithm_simd {
                         .as_mut_ptr()
                         .copy_from_nonoverlapping(input.as_ptr(), to_copy);
                     let simd_input = SimdInput::new(&tmpbuf.0);
-                    algorithm.check_utf8(simd_input);
                     idx += to_copy;
+                    if unlikely!(!simd_input.is_ascii()) {
```

Comment:
Would it not be better to let the branch predictor decide whether these ASCII-only branches are likely or unlikely? Do we know what the effect of this is on CJK text for example?

Reply:
I am doubtful as well. It only has an effect with nightly Rust and the hint features enabled anyway. I will run some benchmarks on aarch64 and x86 and throw it out if there is no measurable benefit.